cacerts/java.bzl

```diff
@@ -0,0 +1,71 @@
+# Copyright 2017 Google Inc. All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Rule for building and extracting java ca-certificates inside of a docker image."""
+
+def _impl(ctx):
```

Comment:
I'd like to come back to my old idea of a "docker_run" rule that supports this general pattern of "run a certain command in a docker image, and extract some paths as outputs. We could use it here and in a few other cases.

Reply:
Holy package list, Batman....
```
  ca-certificates dbus fontconfig-config fonts-dejavu-core java-common
  krb5-locales libalgorithm-c3-perl libarchive-extract-perl libasyncns0
  libavahi-client3 libavahi-common-data libavahi-common3 libcap-ng0
  libcgi-fast-perl libcgi-pm-perl libclass-c3-perl libclass-c3-xs-perl
  libcpan-meta-perl libcups2 libdata-optlist-perl libdata-section-perl
  libdbus-1-3 libexpat1 libfcgi-perl libffi6 libflac8 libfontconfig1
  libfreetype6 libgdbm3 libglib2.0-0 libglib2.0-data libgmp10
  libgnutls-deb0-28 libgssapi-krb5-2 libhogweed2 libice6 libjpeg62-turbo
  libjson-c2 libk5crypto3 libkeyutils1 libkrb5-3 libkrb5support0 liblcms2-2
  liblog-message-perl liblog-message-simple-perl libmodule-build-perl
  libmodule-pluggable-perl libmodule-signature-perl libmro-compat-perl
  libnettle4 libnspr4 libnss3 libogg0 libp11-kit0 libpackage-constants-perl
  libparams-util-perl libpcsclite1 libpng12-0 libpod-latex-perl
  libpod-readme-perl libpulse0 libregexp-common-perl libsctp1 libsm6
  libsndfile1 libsoftware-license-perl libsqlite3-0 libssl1.0.0
  libsub-exporter-perl libsub-install-perl libtasn1-6 libterm-ui-perl
  libtext-soundex-perl libtext-template-perl libvorbis0a libvorbisenc2
  libwrap0 libx11-6 libx11-data libx11-xcb1 libxau6 libxcb1 libxdmcp6 libxext6
  libxi6 libxml2 libxtst6 lksctp-tools openjdk-7-jre-headless openssl perl
  perl-base perl-modules rename sgml-base shared-mime-info tcpd tzdata-java
  ucf x11-common xdg-user-dirs xml-core
```

OMG.

Reply:
I think what you are describing is effectively `genrule` where the command is invoked within a specified container image; I think we should strive to keep the signature as close to `genrule` as possible if we head this route (I wonder if we'd want dicts for the inputs / outputs?).  It would certainly make a useful building block for this.

Reply:
Yeah, that's basically what I was thinking. I have a few other places I think we'll need to implement a rule like this, and a docker_genrule would definitely help.